onnxruntime/core/providers/cuda/generator/range.cc

```diff
@@ -49,15 +49,16 @@ static Status ComputeRange(OpKernelContext* ctx) {
 
   // Start, Limit and Delta are stored in GPU. So we need copy it to CPU to read.
   // It is better to store these tensors in pinned memory or CPU for better performance.
+  // cudaMemcpyAsync from device memory to pageable host memory will return only once the copy has completed.
```

Comment:
>// cudaMemcpyAsync from device memory to pageable host memory will return only once the copy has completed. [](start = 2, length = 107)

Seems some other places there are still explicit cudaStreamSynchronize()

Reply:
some other places are probably from pageable host memory to device memory. 

> > // cudaMemcpyAsync from device memory to pageable host memory will return only once the copy has completed. [](start = 2, length = 107)
> 
> Seems some other places there are still explicit cudaStreamSynchronize()

some other places are probably from pageable host memory to device memory.